src/dataframes.jl

```diff
@@ -21,12 +21,31 @@ function Plots.extractGroupArgs(group::Symbol, df::AbstractDataFrame, args...)
     Plots.extractGroupArgs(collect(df[group]))
 end
 
+# allows the passing of expressions including DataFrame columns as symbols
+function processExpr(expr, df)
+    arg = map(expr.args) do x
+        isa(x, Expr) && return processExpr(x, df)
+        if isa(x, QuoteNode) && isa(x.value, Symbol)
+            return collect(df[x.value])
```

Comment:
Hopefully this example clarifies what you're doing:

```julia
julia> ex = :(:x + 1)
:(:x + 1)

julia> dump(ex)
Expr
  head: Symbol call
  args: Array{Any}((3,))
    1: Symbol +
    2: QuoteNode
      value: Symbol x
    3: Int64 1
  typ: Any

julia> ex.args[2] = rand(10)
10-element Array{Float64,1}:
 0.573531 
 0.0145821
 0.531146 
 0.683965 
 0.439694 
 0.24342  
 0.314299 
 0.540411 
 0.996641 
 0.346173 

julia> dump(ex)
Expr
  head: Symbol call
  args: Array{Any}((3,))
    1: Symbol +
    2: Array{Float64}((10,)) [0.573531,0.0145821,0.531146,0.683965,0.439694,0.24342,0.314299,0.540411,0.996641,0.346173]
    3: Int64 1
  typ: Any

julia> eval(ex)
10-element Array{Float64,1}:
 1.57353
 1.01458
 1.53115
 1.68397
 1.43969
 1.24342
 1.3143 
 1.54041
 1.99664
 1.34617
```

It works, but you're overworking `eval`